lib/srv/authhandlers.go

```diff
@@ -273,9 +275,28 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 	}
 	teleportUser := cert.KeyId
 
+	connectionDiagnosticID := cert.Extensions[teleport.CertExtensionConnectionDiagnosticID]
+
 	// only failed attempts are logged right now
 	recordFailedLogin := func(err error) {
 		failedLoginCount.Inc()
+
+		message := "Principal is not allowed by this certificate. Ensure your roles allows you use it."
+		traceType := types.ConnectionDiagnosticTrace_RBAC_PRINCIPAL
+
+		if trace.IsAccessDenied(err) {
+			message = "You are not authorized to access this node. Ensure your role allows you to access it."
```

Comment:
What's the difference between "principal is not allowed" error above and "you are not authorized" error here? In which cases does the one vs the other happen?

Reply:
Principal not allowed happens when the requested principal is not part of the allowed login.

This one is when the node is not accessible because of `node_labels` rules.

I'll try to use a better description for both messages by saying which field should be changed to allow access.